File: tests/fixtures/openai-realtime/openai-realtime-reconnect-on-drop-broken/02-noop-close.ts
import WebSocket from 'ws';

class AudioInterceptor {
  #agentOpenAISocket?: WebSocket;
  private closedAt?: number;

  setup() {
    const url = 'wss://api.openai.com/v1/realtime?model=gpt-realtime';
    this.#agentOpenAISocket = new WebSocket(url, { headers: {} });

    this.#agentOpenAISocket.on('close', () => {
      this.closedAt = Date.now();
    });
  }
}

export default AudioInterceptor;

File: tests/fixtures/openai-realtime/openai-realtime-reconnect-on-drop-broken/02_noop_close.py
import websockets

WS_URL = "wss://api.openai.com/v1/realtime?model=gpt-realtime"


async def stream():
    ws = await websockets.connect(WS_URL)
    try:
        async for message in ws:
            handle_message(message)
    except websockets.exceptions.ConnectionClosedError:
        pass
